packages/core/test/result-transformers.test.ts

```diff
@@ -268,6 +268,153 @@ describe('resultTransformers', () => {
     })
   })
 
+  describe('.summary()', () => {
+    describe('with a valid result', () => {
+      it('it should return an ResultSummary', async () => {
+        const resultStreamObserverMock = new ResultStreamObserverMock()
+        const query = 'Query'
+        const params = { a: 1 }
+        const meta = { db: 'adb' }
+        const result = new Result(Promise.resolve(resultStreamObserverMock), query, params)
+        const keys = ['a', 'b']
+        const rawRecord1 = [1, 2]
+        const rawRecord2 = [3, 4]
+        resultStreamObserverMock.onKeys(keys)
+        resultStreamObserverMock.onNext(rawRecord1)
+        resultStreamObserverMock.onNext(rawRecord2)
+        resultStreamObserverMock.onCompleted(meta)
+
+        const summary: ResultSummary = await resultTransformers.summary()(result)
+
+        expect(summary).toEqual(
+          new ResultSummary(query, params, meta)
+        )
+      })
+
+      it('it should cancel stream', async () => {
+        const meta = { db: 'adb' }
+        const resultStreamObserverMock = new ResultStreamObserverMock()
+        const cancelSpy = jest.spyOn(resultStreamObserverMock, 'cancel')
+        cancelSpy.mockImplementation(() => resultStreamObserverMock.onCompleted(meta))
+        const query = 'Query'
+        const params = { a: 1 }
+        const result = new Result(Promise.resolve(resultStreamObserverMock), query, params)
+        const keys = ['a', 'b']
+        const rawRecord1 = [1, 2]
+        const rawRecord2 = [3, 4]
+        resultStreamObserverMock.onKeys(keys)
+        resultStreamObserverMock.onNext(rawRecord1)
+        resultStreamObserverMock.onNext(rawRecord2)
+
+        const summary: ResultSummary = await resultTransformers.summary()(result)
+
+        expect(cancelSpy).toHaveBeenCalledTimes(1)
+        expect(summary).toEqual(
+          new ResultSummary(query, params, meta)
+        )
+      })
+
+      it('it should return a ResultSummary<number>', async () => {
+        const resultStreamObserverMock = new ResultStreamObserverMock()
+        const query = 'Query'
+        const params = { a: 1 }
+        const meta = { db: 'adb' }
+        const result = new Result(Promise.resolve(resultStreamObserverMock), query, params)
+        const keys = ['model', 'year']
+        const rawRecord1 = ['Beautiful Sedan', 1987]
+        const rawRecord2 = ['Hot Hatch', 1995]
+
+        resultStreamObserverMock.onKeys(keys)
+        resultStreamObserverMock.onNext(rawRecord1)
+        resultStreamObserverMock.onNext(rawRecord2)
+        resultStreamObserverMock.onCompleted(meta)
+        const summary = await resultTransformers.summary<number>()(result)
```

Comment:
```suggestion
        const summary: ResultSummary<number> = await resultTransformers.summary<number>()(result)
```
Does this make sense to make sure the defined return type is really what you expect it to be?